app/controllers/clients_controller.rb

```diff
@@ -31,22 +31,12 @@ def update
     respond_to do |format|
       if @client.update(client_params)
         format.html { redirect_to @client, notice: 'Client was successfully updated.' }
-        format.json { render :show, status: :ok, location: @client }
       else
-        format.html { render :edit }
-        format.json { render json: @client.errors, status: :unprocessable_entity }
+        format.html { render :edit } unless @client.update(profile_params)
       end
     end
   end
 
-  def destroy
```

Comment:
As far as I know my story depends on destroy, because in my story only the socialworker cannot destroy the client. Admin and superadmin can. [Trello Card - As a Social Worker, I want to register clients](https://trello.com/c/xzzDH6dD)
Or am I getting this wrong?